SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.WindowsDesktop/Utils/ProcessInformation.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.AppCenter.Crashes.Utils
+{
+    public class ProcessInformation : IProcessInformation
+    {
+        public DateTime? ProcessStartTime
+        {
+            get
+            {
+                try
+                {
+                    return Process.GetCurrentProcess().StartTime;
+                }
+                catch (Exception e)
+                {
+                    AppCenterLog.Warn(Crashes.LogTag, "Unable to get process start time.", e);
+                    return null;
+                }
+            }
+        }
+
+        public int? ProcessId
+        {
+            get
+            {
+                try
+                {
+                    return Process.GetCurrentProcess().Id;
+                }
+                catch (Exception e)
+                {
+                    AppCenterLog.Warn(Crashes.LogTag, "Unable to get process ID.", e);
+                    return null;
+                }
+            }
+        }
+
+        public string ProcessName
+        {
+            get
+            {
+                try
+                {
+                    return Process.GetCurrentProcess().ProcessName;
+                }
+                catch (Exception e)
+                {
+                    AppCenterLog.Warn(Crashes.LogTag, "Unable to get process name.", e);
+                    return null;
+                }
+            }
+        }
+
+        // Parent process information is not (easily) available, and not necessary.
```

Comment:
Can you give a bit more context on this?  Did the crashes team say it's not necessary?  How would we get the parent process information if we decided we want it?

Reply:
Yeah they said it is not necessary. The steps I believe involve using C++ code which we really want to avoid.